src/OSPSuite.Core/Domain/Populations/CovariateValuesCache.cs

```diff
@@ -102,8 +102,8 @@ public virtual CovariateValues GetOrAddCovariateValuesFor(string covariateName)
 
       public void Remove(string covariateName) => _covariateValuesCache.Remove(covariateName);
 
-      public string[] AllCovariateValuesForIndividual(int individualId) => AllCovariateValues.Select(x => x.ValueAt(individualId)).ToArray();
+      public string[] AllCovariateValuesForIndividual(int individualIndex) => AllCovariateValues.Select(x => x.ValueAt(individualIndex)).ToArray();
```

Comment:
This was assuming index == id which is wrong. We do not support retrieval by individualId at the moment, only by index